### PR TITLE
Created submit for CaptureWebsiteController

### PR DESCRIPTION
**Pull request by davidmthornton**
A few things I've done:-

1. corrected the messages for ConfirmWebsiteView
2. added a viewspec for it
3. built submit function for CaptureWebsiteController
4. added current email on ConfirmWebsiteView (pulled from session)

### CODE REVIEW

File: test/views/ConfirmWebsiteViewSpec.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    val continueButton = ".button"
+    val editLink = "#content > article > p:nth-child(5) > a"
+  }
+
+  "The Confirm Website view" should {
+    lazy val view = injectedView(testWebsite)
+    lazy implicit val document: Document = Jsoup.parse(view.body)
+
+    "have the correct heading" in {
+      document.getElementsByClass(Selectors.heading).text() shouldBe "Confirm the website address"
```
Comment (lewishay) [inline test/views/ConfirmWebsiteViewSpec.scala]: Use the helper function `elementText`

File: test/views/ConfirmWebsiteViewSpec.scala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      "should have the correct text" in {
+        elementText(Selectors.backLink) shouldBe "Back"
+      }
+
+      "should have the correct back link" in {
+        element(Selectors.backLink).attr("href") shouldBe controllers.routes.CaptureWebsiteController.show().url
+      }
+    }
+
+    "have the website address the user provided" in {
```
Comment (lewishay) [inline test/views/ConfirmWebsiteViewSpec.scala]: This test seems a bit unspecific. There's currently also no test for the paragraph containing the website, so it'd make more sense to test that specific paragraph text which would also test that the website is played back.

File: test/views/ConfirmWebsiteViewSpec.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  object Selectors {
+    val heading = "heading-large"
+    val backLink = "#content > article > a"
+    val continueButton = ".button"
+    val editLink = "#content > article > p:nth-child(5) > a"
+  }
+
+  "The Confirm Website view" should {
+    lazy val view = injectedView(testWebsite)
+    lazy implicit val document: Document = Jsoup.parse(view.body)
```
Comment (lewishay) [inline test/views/ConfirmWebsiteViewSpec.scala]: Could you test the page title (`document.title` shouldBe ...)

File: app/controllers/CaptureWebsiteController.scala [hunk 4]
```diff
@@ -44,12 +43,31 @@ class CaptureWebsiteController @Inject()(val authComps: AuthPredicateComponents,
...
       Future.successful(errorHandler.showInternalServerError)
     }
   }
 
-  def submit: Action[AnyContent] = ???
+  def submit: Action[AnyContent] = (allowAgentPredicate andThen inflightCheck).async { implicit user =>
+    val validationWebsite: Option[String] = user.session.get(SessionKeys.validationWebsiteKey)
+    val prepopulationWebsite: Option[String] = user.session.get(SessionKeys.websiteKey)
+
+    (validationWebsite, prepopulationWebsite) match {
```
[the remark this answers is not in the document]
Comment (davidmthornton) [reply app/controllers/CaptureWebsiteController.scala]: It was just in there temporarily but I can take it out.

File: app/views/ConfirmWebsiteView.scala.html [hunk 5]
```diff
@@ -23,11 +23,13 @@
...
 
- <a class="link-back" href=''>@messages("base.back")</a>
+ <a class="link-back" href="@controllers.routes.CaptureWebsiteController.show().url">@messages("base.back")</a>
 
  <h1 class="heading-large">@messages("confirmWebsite.heading")</h1>
 
- <h2 class="lede">@website</h2>
+ <h2 class="lede">@messages("confirmWebsite.heading", website)</h2>
+
+ <p><a href="@controllers.routes.CaptureWebsiteController.show().url">Change the website address</a></p>
```
[the remark this answers is not in the document]
Comment (davidmthornton) [reply app/views/ConfirmWebsiteView.scala.html]: You mean wrong message? Updated.

File: app/views/ConfirmWebsiteView.scala.html [hunk 6]
```diff
@@ -23,11 +23,13 @@
 
 @mainTemplate(title = messages("confirmWebsite.title")) {
 
- <a class="link-back" href=''>@messages("base.back")</a>
+ <a class="link-back" href="@controllers.routes.CaptureWebsiteController.show().url">@messages("base.back")</a>
 
  <h1 class="heading-large">@messages("confirmWebsite.heading")</h1>
 
- <h2 class="lede">@website</h2>
+ <h2 class="lede">@messages("confirmWebsite.heading", website)</h2>
```
[the remark this answers is not in the document]
Comment (davidmthornton) [reply app/views/ConfirmWebsiteView.scala.html]: updated

File: test/views/ConfirmWebsiteViewSpec.scala (hunk 1 above)
Comment (davidmthornton) [reply test/views/ConfirmWebsiteViewSpec.scala]: Done

File: app/views/ConfirmWebsiteView.scala.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (lewishay) [reply app/views/ConfirmWebsiteView.scala.html]: I mean you aren't using a message here at all, you have the text `Change the website address` hardcoded in the `<a>` tag

File: app/views/ConfirmWebsiteView.scala.html [hunk 7]
```diff
@@ -23,11 +23,13 @@
 
 @mainTemplate(title = messages("confirmWebsite.title")) {
 
- <a class="link-back" href=''>@messages("base.back")</a>
+ <a class="link-back" href="@controllers.routes.CaptureWebsiteController.show().url">@messages("base.back")</a>
 
  <h1 class="heading-large">@messages("confirmWebsite.heading")</h1>
 
- <h2 class="lede">@website</h2>
+ <p class="lede">@messages("confirmWebsite.p1", website)</p>
```
Comment (lewishay) [inline app/views/ConfirmWebsiteView.scala.html]: It isn't using a `lede` class in the prototype either

File: app/common/SessionKeys.scala [hunk 8]
```diff
@@ -22,6 +22,7 @@ object SessionKeys {
   val validationEmailKey = "validationEmail"
   val prepopulationWebsiteKey = "vatCorrespondencePrepoulationWebsite"
   val validationWebsiteKey = "validationWebsite"
+  val websiteKey = "Website"
```
Comment (lewishay) [inline app/common/SessionKeys.scala]: Do we need this extra session key? AFAIK we only need the two keys above for the website; one for the form validation and one for pre-pop. When you submit the form then it should add the value to the pre-pop key.
While you're here, would you mind updating the `validationWebsiteKey` value to be `vatCorrespondenceValidationWebsite`? We were pulled up on having session keys that weren't specific enough before.

**Pull request merged**